Block/Adminhtml/Usage/Edit/CustomLicense.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace DevStone\UsageCalculator\Block\Adminhtml\Usage\Edit;
+
+/**
+ * Class CustomLicense
+ * @package DevStone\UsageCalculator\Block\Adminhtml\Usage\Edit
+ */
+class CustomLicense implements \Magento\Framework\View\Element\UiComponent\Control\ButtonProviderInterface
+{
+    /**
+     * @var \Magento\Framework\UrlInterface
+     */
+    protected $urlBuilder;
+
+    /**
+     * CustomLicense constructor.
+     * @param \Magento\Framework\UrlInterface $urlBuilder
+     */
+    public function __construct(
+        \Magento\Framework\UrlInterface $urlBuilder
+    ) {
+        $this->urlBuilder = $urlBuilder;
+    }
+
+    /**
+     * @return array
+     */
+    public function getButtonData()
+    {
+        return [
+            'label' => __('Add Custom License'),
+            'on_click' => sprintf("location.href = '%s';", $this->getCustomLicense(['custom_license' => true])),
```

Comment:
Why you using `on_click` here???? There should be no need for JS and using the `href` attribute is much preferred.

Reply:
The `custom_license` parameter also needs to be added to any existing custom licenses the admin wishes to edit when they select it in the list of usages.

Reply:
yes, custom_license parameter is added to existing custom licenses.

Reply:
> Why you using `on_click` here???? There should be no need for JS and using the `href` attribute is much preferred.
 
Magento-core also using this method on-click while creating a button using block [Reference](https://github.com/magento/magento2/blob/abbfa03e3091ba335fbad99275667265f966e516/app/code/Magento/Cms/Block/Adminhtml/Block/Edit/BackButton.php#L22)